packages/fast-components-react-base/src/slider/slider.tsx

```diff
@@ -0,0 +1,1078 @@
+            (this.props.mode === SliderMode.singleValue &&
+                thumb === SliderThumb.lowerThumb)
+        ) {
+            return;
+        }
+
+        if (typeof this.props.thumb === "function") {
+            return this.props.thumb(
+                this.props,
+                this.state,
+                thumb === SliderThumb.upperThumb
+                    ? this.handleUpperThumbMouseDown
+                    : this.handleLowerThumbMouseDown,
+                thumb === SliderThumb.upperThumb
+                    ? this.handleUpperThumbKeyDown
+                    : this.handleLowerThumbKeyDown,
+                thumb
+            );
+        } else {
+            return this.renderDefaultThumb(
+                this.props,
+                this.state,
+                thumb === SliderThumb.upperThumb
+                    ? this.handleUpperThumbMouseDown
+                    : this.handleLowerThumbMouseDown,
+                thumb === SliderThumb.upperThumb
+                    ? this.handleUpperThumbKeyDown
+                    : this.handleLowerThumbKeyDown,
+                thumb
+            );
+        }
+    }
+
+    /**
+     *  Renders the default thumb
+     */
+    private renderDefaultThumb(
+        props: SliderProps,
+        state: SliderState,
+        mouseDownCallback: (event: React.MouseEvent) => void,
+        keyDownCallback: (event: React.KeyboardEvent) => void,
+        thumb: SliderThumb
+    ): React.ReactNode {
+        return (
+            <SliderTrackItem
+                {...this.getThumbManagedClasses(thumb)}
+                minValuePositionBinding={
+                    thumb === SliderThumb.lowerThumb
+                        ? SliderTrackItemAnchor.selectedRangeMin
+                        : undefined
+                }
+                maxValuePositionBinding={
+                    thumb === SliderThumb.upperThumb
+                        ? SliderTrackItemAnchor.selectedRangeMax
+                        : undefined
+                }
+                draggable={false}
+                tabIndex={0}
+                onKeyDown={keyDownCallback}
+                onMouseDown={mouseDownCallback}
+                aria-valuemin={props.range.minValue}
+                aria-valuemax={props.range.maxValue}
+                aria-valuenow={
+                    thumb === SliderThumb.lowerThumb ? state.lowerValue : state.upperValue
+                }
+                aria-labelledby={props.labelledBy || null}
+            />
+        );
+    }
+
+    /**
+     * Renders a hidden input element which can interact with a
+     * form hosting this component
+     */
+    private renderHiddenInputElement(): React.ReactNode {
+        let formattedValue: string = "";
+
+        switch (this.props.mode) {
+            case SliderMode.adjustBoth:
+                formattedValue =
+                    "[" + this.state.lowerValue + "," + this.state.upperValue + "]";
```

Comment:
can you use template literal syntax here

Reply:
yep